### PR TITLE
solution

### PR DESCRIPTION
**Pull request by savchuksavchuk**
[DEMO LINK](https://savchuksavchuk.github.io/react_movies-list-add-form/)

### CODE REVIEW

File: src/components/NewMovie/NewMovie.tsx [hunk 1]
```diff
@@ -1,45 +1,106 @@
...
+    onAdd(newMovie);
+    setCount(prevCount => prevCount + 1);
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  // eslint-disable-next-line
```
Comment (loralevitska) [inline src/components/NewMovie/NewMovie.tsx:46]: try to fix this linter rule and remove comment

File: src/components/NewMovie/NewMovie.tsx [hunk 2]
```diff
@@ -1,45 +1,106 @@
...
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => {
+        event.preventDefault();
+
+        handlerOnAdd();
+      }}
```
Comment (loralevitska) [inline src/components/NewMovie/NewMovie.tsx:61]: move this function to variable and use only function name for `onSubmit` property. Pay attention to the code style, fix it all over the filles

**Review by loralevitska — CHANGES_REQUESTED** [review]
hi, good job!  need some improvements
1. After lose focus url input show errors although inputs have value
image

**Comment by savchuksavchuk** [comment]
Hello. The error after unfocus ImageUrl and ImdbUrl - it's the url validation. I'm not shure, if it should work after unfocus or after submit but I added 'URL is invalid' error for this validation. I think i will be less confusing.
About regexp - I have no idea how to fix it. I tried to create regexp with new Regexp constructor - but it causes a lot of mistakes.

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,45 +1,104 @@
...
+  const patterValidation = (string: string) => {
+    return pattern.test(string);
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => handlerOnAdd(event)}
```
Comment (TarasHoliuk) [inline src/components/NewMovie/NewMovie.tsx:59]:
```suggestion
      onSubmit={handlerOnAdd}
```

File: src/components/TextField/TextField.tsx [hunk 4]
```diff
@@ -19,14 +20,20 @@ export const TextField: React.FC<Props> = ({
...
   onChange = () => {},
+  onPatternValidation = () => true,
+
 }) => {
-  // generage a unique id once on component load
   const [id] = useState(() => `${name}-${getRandomDigits()}`);
-
-  // To show errors only if the field was touched (onBlur)
   const [touched, setToched] = useState(false);
+  const [isPatternValid, setPatternValidation] = useState(true);
```
Comment (TarasHoliuk) [inline src/components/TextField/TextField.tsx:28]:
```suggestion
  const [isPatternValid, setIsPatternValid] = useState(true);
```

File: src/components/TextField/TextField.tsx [hunk 5]
```diff
@@ -19,14 +20,20 @@ export const TextField: React.FC<Props> = ({
...
+
   const hasError = touched && required && !value;
 
+  const onBlurHandler = () => {
+    setToched(true);
+    setPatternValidation(() => onPatternValidation(value));
```
Comment (TarasHoliuk) [inline src/components/TextField/TextField.tsx:34]:
```suggestion
    setPatternValidation(onPatternValidation(value));
```

**Comment by TarasHoliuk** [comment]
> Hello. The error after unfocus ImageUrl and ImdbUrl - it's the url validation. I'm not shure, if it should work after unfocus or after submit but I added 'URL is invalid' error for this validation. I think i will be less confusing. About regexp - I have no idea how to fix it. I tried to create regexp with new Regexp constructor - but it causes a lot of mistakes.

The possible solution is show to 'URL is invalid' if the input is also not empty

**Review by TarasHoliuk — CHANGES_REQUESTED** [review]

**Comment by savchuksavchuk** [comment]
Hello. Now it works like this:
1. If url input is empty - 'URL is required' is showing
2. If url is not empty but it isn't correct 'URL is invalid' is showing

About the failed tests - I watched screenshot and it shows 'URL is invalid' error. I think tests are not provided for this validation but actually it works correctly.

Also I added the disabling for submit button if URL is invalid.

image
image

**Review by polosanya — APPROVED** [review]
Good job!

**Pull request closed without merging**